src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,122 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int MAX_CAPACITY = Integer.MAX_VALUE;
+    private Node<K, V>[] table = new Node[INITIAL_CAPACITY];
+    private int size;
+    private int threshold;
+    private int capacity;
 
     @Override
     public void put(K key, V value) {
+        Node<K, V> newNode = new Node<>(key == null ? 0
+                : key.hashCode(), key, value, null);
+        capacity = table.length;
+        int position = getPosition(key, capacity);
+        threshold = (int) (LOAD_FACTOR * capacity);
+
+        if (size >= threshold) {
+            table = resize();
+        }
+
+        for (int i = 0; i < capacity; i++) {
+            if (table[i] != null && Math.abs(table[i].key == null ? 0
+                    : table[i].hash % capacity) == position) {
+                Node<K, V> node = table[i];
+                while (node != null) {
+                    if (Objects.deepEquals(node.key, key)) {
+                        node.value = value;
+                        return;
+
+                    } else if (node.next == null) {
+                        node.next = newNode;
+                        size++;
+                        return;
+                    }
+                    node = node.next;
+                }
+            }
+        }
 
+        table[position] = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        if (capacity == 0) {
+            return null;
+        }
+        int position = getPosition(key, capacity);
+        for (int i = 0; i < capacity; i++) {
+            if (table[i] != null && i == position) {
+                Node<K, V> node = table[i];
+                for (int j = 0; j <= size; j++) {
+                    if (node.key == null && key == null || Objects.equals(node.key, key)) {
```

Comment:
Objects#equals has null check inside, so you can simplify condition to
`if (Objects.equals(...))`